rocAL/rocAL/source/api/rocal_api_meta_data.cpp

```diff
@@ -345,27 +306,69 @@ ROCAL_API_CALL rocalGetOneHotImageLabels(RocalContext p_context, void* buf, int
 }
 
 
-void
-ROCAL_API_CALL rocalGetBoundingBoxCords(RocalContext p_context, float* buf)
+RocalTensorList
+ROCAL_API_CALL rocalGetBoundingBoxCords(RocalContext p_context)
 {
     if (!p_context)
         THROW("Invalid rocal context passed to rocalGetBoundingBoxCords")
     auto context = static_cast<Context*>(p_context);
+    return context->master_graph->bbox_meta_data();
+}
+
+unsigned
+ROCAL_API_CALL rocalGetMaskCount(RocalContext p_context, int* buf)
+{
+    if (p_context == nullptr)
+        THROW("Invalid rocal context passed to rocalGetMaskCount")
+    unsigned size = 0, count = 0;
+    auto context = static_cast<Context*>(p_context);
     auto meta_data = context->master_graph->meta_data();
-    size_t meta_data_batch_size = meta_data.second->get_bb_cords_batch().size();
+    size_t meta_data_batch_size = meta_data.second->get_mask_cords_batch().size();
     if(context->user_batch_size() != meta_data_batch_size)
         THROW("meta data batch size is wrong " + TOSTR(meta_data_batch_size) + " != "+ TOSTR(context->user_batch_size() ))
     if(!meta_data.second)
+        THROW("No mask has been loaded for this output image")
+    for(unsigned i = 0; i < meta_data_batch_size; i++)
     {
-        WRN("No label has been loaded for this output image")
-        return;
+        unsigned object_count = meta_data.second->get_labels_batch()[i].size();
+        for(unsigned int j = 0; j < object_count; j++) {
+            unsigned polygon_count = meta_data.second->get_mask_polygons_count_batch()[i][j];
+            buf[count++] = polygon_count;
+            size += polygon_count;
+        }
     }
-    for(unsigned i = 0; i < meta_data_batch_size; i++)
+    return size;
+}
+
+RocalTensorList
+ROCAL_API_CALL rocalGetMaskCoordinates(RocalContext p_context, int *bufcount)
+{
+    if (p_context == nullptr)
+        THROW("Invalid rocal context passed to rocalGetMaskCoordinates")
+    auto context = static_cast<Context*>(p_context);
+    auto meta_data = context->master_graph->meta_data();
+    size_t meta_data_batch_size = meta_data.second->get_mask_cords_batch().size();
+    if(context->user_batch_size() != meta_data_batch_size)
+        THROW("meta data batch size is wrong " + TOSTR(meta_data_batch_size) + " != "+ TOSTR(context->user_batch_size() ))
+    if(!meta_data.second)
+        THROW("No mask has been loaded for this output image")
+    int size = 0;
+    for(unsigned image_idx = 0; image_idx < meta_data_batch_size; image_idx++)
     {
-        unsigned bb_count = meta_data.second->get_bb_cords_batch()[i].size();
-        memcpy(buf, meta_data.second->get_bb_cords_batch()[i].data(), bb_count * sizeof(BoundingBoxCord));
-        buf += (bb_count * 4);
+        int poly_size = 0;
```

Comment:
Is poly_size variable used anywhere? Its just getting incremented and not used anywhere. Check and remove if not needed

Reply:
Removed